pkg/manage/server.go

```diff
@@ -19,14 +22,19 @@ func RunServer(cfg *ServerConfig) error {
 		return fmt.Errorf("listen on addr %s: %w", addr, err)
 	}
 
-	s := grpc.NewServer()
-	proto.RegisterManageServer(s, newServer(cfg))
+	srv := grpc.NewServer()
+	proto.RegisterManageServer(srv, newServer(cfg))
 
-	fmt.Printf("Running manage service on %s\n", addr)
+	go func() {
+		waitForShutdown()
+		srv.GracefulStop()
```

Comment:
Is it possible that the process exits regularly (in the main()) before this is done?

Reply:
No. Why do you think that?